server/status.go

```diff
@@ -39,68 +38,73 @@ import (
-	statusTransactionsKeyPrefix = statusKeyPrefix + "txns/"
-
+	/*
+		Note that :node_id can always be replaced by the value "local" to see
+		the local nodes response.
+
+		/_status/details/:node_id		 - specific node's details
+		/_status/gossip/:node_id         - specific node's gossip
+		/_status/logfiles/:node_id       - list log files
+		/_status/logfiles/:node_id/:file - returns the contents of the specific
+										   log files on specific node
+		/_status/logs/:node_id           - log entries from a specific node
+		/_status/stacks/:node_id		 - exposes stack traces of running
+										   goroutines
+		/_status/nodes				     - all nodes' status
+		/_status/nodes/:node_id		     - a specific node's status
+		/_status/stores                  - all stores' status
+		/_status/stores/:store_id        - a specific store's status
+	*/
+
+	// statusPrefix is the root of the cluster statistics and metrics API.
+	statusPrefix = "/_status/"
+
+	// statusGossipPattern exposes a view of the gossip network.
+	statusGossipPattern = "/_status/gossip/:node_id"
+	statusGossipLocal   = "/_status/gossip/local"
+
+	// statusDetailsPattern exposes a node's details.
+	statusDetailsPattern = "/_status/details/:node_id"
+	statusDetailsLocal   = "/_status/details/local"
+
+	// statusLogFilesListPattern exposes a list of log files.
+	statusLogFilesListPattern = "/_status/logfiles/:node_id"
+	statusLogFilesListLocal   = "/_status/logfiles/local"
+	// statusLogFilePattern exposes a specific file on a node.
+	statusLogFilePattern = "/_status/logfiles/:node_id/:file"
+
+	// statusLogKeyPrefix exposes the logs for each node.
+	statusLogsPattern = "/_status/logs/:node_id"
+	statusLogsLocal   = "/_status/logs/local"
 	// Default Maximum number of log entries returned.
 	defaultMaxLogEntries = 1000
 
-	// Timeout for used for looking up log entries in another node.
-	logEntriesTimeout = time.Second
+	// statusStacksPattern exposes the stack traces of running goroutines.
+	statusStacksPattern = "/_status/stacks/:node_id"
+	statusStacksLocal   = "/_status/stacks/local"
+	// stackTraceApproxSize is the approximate size of a goroutine stack trace.
+	stackTraceApproxSize = 1024
+
+	// statusNodesPattern exposes status for all nodes in the cluster.
+	statusNodesPattern = "/_status/nodes/"
+	// statusNodePattern exposes status for a single node.
+	statusNodePattern = "/_status/nodes/:node_id"
+	statusNodeLocal   = "/_status/nodes/local"
+
+	// statusStoresPattern exposes status for all stores in the cluster.
+	statusStoresPattern = "/_status/stores/"
+	// statusStorePattern exposes status for a single store.
+	statusStorePattern = "/_status/stores/:store_id"
+	statusStoreLocal   = "/_status/stores/local"
+
+	// statusProxyTimeout is the timeout used when proxying a request to another
+	// node.
+	statusProxyTimeout = time.Second
 )
 
+// Pattern for local used when determining the node ID.
+var localRE = regexp.MustCompile(`(?i)local`)
```

Comment:
case insensitive.
https://stackoverflow.com/questions/15326421/how-do-i-do-a-case-insensitive-regular-expression-in-go